File: Rotate_Array.cpp
#include <vector>
#include <algorithm> // for std::reverse

class Solution {
public:
    void rotate(vector<int>& arr, int k) {
        int n = arr.size();
        k = k % n; // to handle cases where k > n
        if (k == 0) return; // no rotation needed if k is 0
        
     
        reverse(arr.begin(), arr.end());
        // Step 2: Reverse the first k elements
        reverse(arr.begin(), arr.begin() + k);
        // Step 3: Reverse the remaining n - k elements
        reverse(arr.begin() + k, arr.end());
    }
};

File: Rotate_Array.java
import java.util.*;
class Solution {

    public static void reverse(int nums[],int s,int e){
        while(s<=e){
            int temp ;
            temp =nums[s];
            nums[s] = nums[e];
            nums[e] = temp;
            s++;
            e--;
        }
    }

    public void rotate(int[] nums, int k) {
        k = k % (nums.length);
        reverse(nums ,0,nums.length-1-k);
        reverse(nums ,nums.length - k,nums.length-1);
        reverse(nums , 0,nums.length-1);

    }
}